packages/runtime/container-runtime/src/dataStores.ts

```diff
@@ -658,7 +658,14 @@ export class DataStores implements IDisposable {
 						trackState,
 						telemetryContext,
 					);
-					summaryBuilder.addWithStats(contextId, contextSummary);
+					// Add the context's summary to the summary tree. Also, prefix all handles in its summary tree with
+					// "channelsTreeName" name and its id since that is where its previous summary would be. The context
+					// only adds the handle id relative to its path.
+					summaryBuilder.addWithStatsAndPrefixHandles(
```

Comment:
Can we actually do this only once when we are at the container runtime and traverse the summary tree once? Otherwise, this might become a performance bottleneck. This should be an HN algorithm to add the paths back. Where H is the height of the tree and N is the number of nodes in the tree. That way we only make it a N algorithm to add the paths back.

Reply:
I thought about that and it would not be hard to do that. However, I am not sure if it gives us a lot of performance benefit. The tree height is mostly fixed to 3 (container runtime -> data store -> DDS). Sure, a DDS can choose to put its data in a big sub-tree but I would not expect to be to be so big that it causes performance degradation.

Reply:
Ok, sounds good to me. Feel free to resolve

Reply:
I'd personally do one full scan at container runtime layer - I think it's more appropriate.
It's more about composability - as it stands, this pattern has to be repeated at every level, which makes it N^2.

That said, need to be careful - need to search repo to ensure everything is correct. For example, see Tree DDS & using builder.addHandle() is an example.

How are you planning to do this change and not break back-compat requirements across layers?